src/StoreKeeper/WooCommerce/B2C/Imports/ShippingMethodImport.php

```diff
@@ -0,0 +1,309 @@
+{
+    use ConsoleProgressBarTrait;
+
+    const SHIPPING_ZONE_NAME_PREFIX = 'SK_';
+    const SHIPPING_LOCATION_TYPE = 'country';
+
+    const SHIPPING_CLASS_FLAT_RATE = 'flat_rate';
+    const SHIPPING_CLASS_FREE_SHIPPING = 'free_shipping';
+    const SHIPPING_CLASS_LOCAL_PICKUP = 'local_pickup';
+
+    const FREE_SHIPPING_REQUIRES = 'min_amount';
+
+    const SK_SHIPPING_TYPE_ALIAS_PARCEL = 'Parcel';
+    const SK_SHIPPING_TYPE_ALIAS_TRUCK_DELIVERY = 'TruckDelivery';
+    const SK_SHIPPING_TYPE_ALIAS_PICKUP_AT_STORE = 'PickupAtStore';
+
+    const SK_SHIPPING_TYPE_MODULE = 'ShippingModule';
+
+    private int $storekeeper_id = 0;
+
+    public function __construct(array $settings = [])
+    {
+        $this->storekeeper_id = array_key_exists('storekeeper_id', $settings) ? (int) $settings['storekeeper_id'] : 0;
+        unset($settings['storekeeper_id']);
+        parent::__construct($settings);
+    }
+
+    protected function getModule()
+    {
+        return 'ShopModule';
+    }
+
+    protected function getFunction()
+    {
+        return 'listShippingMethodsForHooks';
+    }
+
+    protected function getFilters()
+    {
+        $filters = [
+            [
+                'name' => 'enabled__=',
+                'val' => '1',
+            ],
+            [
+                'name' => 'is_system__=',
+                'val' => '0',
+            ],
+        ];
+
+        if ($this->storekeeper_id > 0) {
+            $filters[] = [
+                'name' => 'id__=',
+                'val' => $this->storekeeper_id,
+            ];
+        }
+
+        return $filters;
+    }
+
+    protected function getLanguage()
+    {
+        return null;
+    }
+
+    /**
+     * @throws ShippingMethodImportException
+     * @throws WordpressException
+     */
+    protected function processItem(Dot $dotObject, array $options = [])
+    {
+        $this->debug('Processing shipping method', $dotObject->get());
+        $storekeeperId = $dotObject->get('id');
+
+        if ($this->isShippingTypeValid($dotObject)) {
+            $this->processShippingMethod($dotObject, $storekeeperId);
+        }
+
+        $this->debug("Unsupported shipping type {$dotObject->get('shipping_type.alias')}, will be skipped.", [
+            'storeKeeperId' => $storekeeperId,
+        ]);
+    }
+
+    private function processShippingMethod(Dot $dotObject, $storekeeperId): void
+    {
+        $countryIso2s = $dotObject->has('country_iso2s') ? $dotObject->get('country_iso2s') : [];
+
+        if (empty($countryIso2s)) {
+            $ShopModule = $this->storekeeper_api->getModule('ShopModule');
+            $response = $ShopModule->getShopWithRelation();
+            $shopData = new Dot($response);
+            $defaultCountryIso2 = $shopData->get('relation_data.business_data.country_iso2');
+            $countryIso2s[] = $defaultCountryIso2;
+        }
+
+        $this->removeOrphanedShippingZones($storekeeperId, $countryIso2s);
```

Comment:
Not sure what you mean here. This `removeOrphanedShippingZones` removes the shipping zones that no longer have any shipping methods because of a country change.

There will be no such thing as orphaned shipping method I believe.